src/lookAheadSuggestion.ts

```diff
@@ -49,7 +49,10 @@ export async function getLookAheadSuggestion(
     2
   );
 
-  const result = findMostRelevantSuggestion(response, completionInfo);
+  const result = findMostRelevantSuggestion(
+    response,
+    getCompletionInfoWithoutOverlappingDot(completionInfo)
```

Comment:
i think its more concise to pass the `completionInfo` object here as is, and call `getCompletionInfoWithoutOverlappingDot` inside `findMostRelevantSuggestion` - as this logic is relevant only to this function.